src/test/java/ru/yandex/praktikum/LoginTests.java

```diff
@@ -0,0 +1,87 @@
+package ru.yandex.praktikum;
+import com.codeborne.selenide.Configuration;
+import com.codeborne.selenide.Selenide;
+import io.qameta.allure.Description;
+import io.qameta.allure.junit4.DisplayName;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import ru.yandex.praktikum.page_object.*;
+
+import static com.codeborne.selenide.Configuration.browser;
+import static com.codeborne.selenide.Selenide.*;
+import static com.codeborne.selenide.WebDriverRunner.url;
+import static org.junit.Assert.assertEquals;
+import static ru.yandex.praktikum.page_object.BaseUrls.*;
+
+
+public class LoginTests {
+
+    MainPage mainPage;
+
+    @Before
```

Comment:
блоки @Before и @After у тебя повторяются в каждом тесте. Нужно  общий код вынести и поместить в одном месте, чтобы не было дублирования кода. В таком случае код поддерживать проще.

Как это можно сделать: создать класс BaseTest, в него вынести этот код, остальные тесты наследовать от BaseTest

Reply:
Didn't create BaseTest as of now because there are two test classes where I don't use 'After' section at all.
Also I added mainPage creation in one of 'Before' sections, so I don't need to duplicate this code for each of the tests in the class.
If the solution recommended is the best pratice for my case though, I can use it in next 'Fixes' iteration